File: semester 4/MPP/JFTS/FestivalPersistance/src/main/java/ro/mpp/repository/DBRepository/UserRepository.java
package ro.mpp.repository.DBRepository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp.domain.User;
import ro.mpp.repository.IUserRepository;
import ro.mpp.utils.Database;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;
import java.util.Optional;

public class UserRepository implements IUserRepository {
    private final Database db;
    private static final Logger logger = LogManager.getLogger(UserRepository.class);

    public UserRepository(Database database) {
        this.db = database;
    }

    @Override
    public Optional<User> findByUsernameAndPassword(String username, String password) {
        logger.info("Finding user {} with password {}", username, password);
        String sql = "SELECT * FROM user WHERE username = ? AND password = ?";

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement stmt = conn.prepareStatement(sql)) {
                stmt.setString(1, username);
                stmt.setString(2, password);
                try (ResultSet rs = stmt.executeQuery()) {
                    if (rs.next()) {
                        return Optional.of(new User(rs));
                    }
                }
            }
        } catch (SQLException e) {
            logger.error("Error in UserRepo::findByUsernameAndPassword", e);
            throw new RuntimeException("Error in UserRepo::findByUsernameAndPassword", e);
        }
        return Optional.empty();
    }

    @Override
    public Optional<User> save(User entity) {
        return Optional.empty();
    }

    @Override
    public Optional<User> update(User entity) {
        return Optional.empty();
    }

    @Override
    public void delete(User entity) {

    }

    @Override
    public Optional<User> find(Integer integer) {
        return Optional.empty();
    }

    @Override
    public List<User> findAll() {
        return List.of();
    }
}

File: semester 4/MPP/C#FTS/C#FTS/Repository/DBRepository/UserRepository.cs
using C_FTS.Domain;
using C_FTS.Utils;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace C_FTS.Repository.DBRepository
{
    public class UserRepository : IUserRepository
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(UserRepository));
        private Database db = new Database();
        public void Delete(User entity)
        {
            throw new NotImplementedException();
        }

        public User? Find(int id)
        {
            throw new NotImplementedException();
        }

        public List<User> FindAll()
        {
            throw new NotImplementedException();
        }

        public User? FindByUsernameAndPassword(string username, string password)
        {
            logger.InfoFormat("Finding user {0} with password {0}", username, password);
            IDbConnection conn = db.GetConnection();

            using (IDbCommand cmd = conn.CreateCommand())
            {
                cmd.CommandText = "SELECT * FROM user WHERE username = @username AND password = @password";
                IDbDataParameter dpName = cmd.CreateParameter();
                dpName.ParameterName = "@username";
                dpName.Value = username;
                cmd.Parameters.Add(dpName);

                IDbDataParameter dpPass = cmd.CreateParameter();
                dpPass.ParameterName = "@password";
                dpPass.Value = password;
                cmd.Parameters.Add(dpPass);

                using (var dataR = cmd.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        User user = new User(dataR);
                        logger.InfoFormat("Exiting Find with {0}", user);
                        return user;
                    }
                    logger.Error("Exiting Find with null");
                    return null;
                }
            }
        }

        public User? Save(User entity)
        {
            throw new NotImplementedException();
        }

        public User? Update(User entity)
        {
            throw new NotImplementedException();
        }
    }
}
